internal/configs/bootnode/bootnode.go

```diff
@@ -0,0 +1,397 @@
+package bootnode
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	nodeconfig "github.com/harmony-one/harmony/internal/configs/node"
+	"github.com/harmony-one/harmony/internal/utils"
+)
+
+// BootNodeConfig contains all the configs user can set for running harmony binary. Served as the bridge
+// from user set flags to internal node configs. Also user can persist this structure to a toml file
+// to avoid inputting all arguments.
+type BootNodeConfig struct {
+	Version    string
+	General    GeneralConfig
+	Network    NetworkConfig
+	P2P        P2pConfig
+	HTTP       HttpConfig
+	WS         WsConfig
+	RPCOpt     RpcOptConfig
+	BLSKeys    BlsConfig
+	Pprof      PprofConfig
+	Log        LogConfig
+	Sys        *SysConfig        `toml:",omitempty"`
+	Devnet     *DevnetConfig     `toml:",omitempty"`
+	Revert     *RevertConfig     `toml:",omitempty"`
+	Legacy     *LegacyConfig     `toml:",omitempty"`
+	Prometheus *PrometheusConfig `toml:",omitempty"`
+	ShardData  ShardDataConfig
+	Cache      CacheConfig
+}
+
+func (bnc BootNodeConfig) ToRPCServerConfig() nodeconfig.RPCServerConfig {
+	readTimeout, err := time.ParseDuration(bnc.HTTP.ReadTimeout)
+	if err != nil {
+		readTimeout, _ = time.ParseDuration(nodeconfig.DefaultHTTPTimeoutRead)
+		utils.Logger().Warn().
+			Str("provided", bnc.HTTP.ReadTimeout).
+			Dur("updated", readTimeout).
+			Msg("Sanitizing invalid http read timeout")
+	}
+	writeTimeout, err := time.ParseDuration(bnc.HTTP.WriteTimeout)
+	if err != nil {
+		writeTimeout, _ = time.ParseDuration(nodeconfig.DefaultHTTPTimeoutWrite)
+		utils.Logger().Warn().
+			Str("provided", bnc.HTTP.WriteTimeout).
+			Dur("updated", writeTimeout).
+			Msg("Sanitizing invalid http write timeout")
+	}
+	idleTimeout, err := time.ParseDuration(bnc.HTTP.IdleTimeout)
+	if err != nil {
+		idleTimeout, _ = time.ParseDuration(nodeconfig.DefaultHTTPTimeoutIdle)
+		utils.Logger().Warn().
+			Str("provided", bnc.HTTP.IdleTimeout).
+			Dur("updated", idleTimeout).
+			Msg("Sanitizing invalid http idle timeout")
+	}
+	evmCallTimeout, err := time.ParseDuration(bnc.RPCOpt.EvmCallTimeout)
+	if err != nil {
+		evmCallTimeout, _ = time.ParseDuration(nodeconfig.DefaultEvmCallTimeout)
+		utils.Logger().Warn().
+			Str("provided", bnc.RPCOpt.EvmCallTimeout).
+			Dur("updated", evmCallTimeout).
+			Msg("Sanitizing invalid evm_call timeout")
+	}
+	return nodeconfig.RPCServerConfig{
+		HTTPEnabled:        bnc.HTTP.Enabled,
```

Comment:
for RPC, we don't need to enable the full list of RPC below right ?

Reply:
These are given from Harmony config.